custom_from.php

```diff
@@ -134,10 +264,88 @@ public function message_compose($params)
                         $score = 1;
                     }
                 }
-            }
+
+                if ($score > 2 && !$should_override_identity) {
+                    break;
+                }
+
+                /** Remove autocomplete by JS frontend after page loaded */
+                $rcmail->output->set_env('identities', array());
+
+                break;
+
+            case self::DEFAULT_IDENTITY_OPTION:
+                $default_identity = $rcmail->user->get_identity();
+                $select_from_attrib = array_intersect_key($attrib, array_flip(array('id', 'class', 'style', 'size', 'tabindex'))) + array(
+                    'name' => '_from',
+                    'onchange' => rcmail_output::JS_OBJECT_NAME . ".change_identity(this)"
+                );
+                $select_from = new html_select($select_from_attrib);
+                $select_from->add(format_email_recipient($default_identity['email'], $default_identity['name']), $default_identity['identity_id']);
+
+                $attrib['content'] = $select_from->show($default_identity['identity_id']);
+
+                break;
         }
 
         self::set_state($compose_id, $address);
+
+        return $attrib;
+    }
+
+    /**
+     * Override headers fields from identity (bcc, reply-to), by plugin settings
+     */
+    public function compose_additional_headers($compose_id, $reply_from, $attrib)
+    {
+        if (!self::get_state($compose_id) && $reply_from !== self::RECEIVING_EMAIL_WITH_DEFAULT_IDENTITY_OPTION) {
+            return $attrib;
+        }
+
+        $part = strtolower($attrib['part']);
+        $value = '';
+
+        switch ($reply_from) {
+            case self::RECEIVING_EMAIL_WITH_DEFAULT_IDENTITY_OPTION:
+                $default_identity = rcmail::get_instance()->user->get_identity();
+                $value = $default_identity[$part] ?? '';
+
+                if ($value === '' && $part == 'replyto') {
+                    $value = $default_identity['reply-to'] ?? '';
+                }
+
+            case self::RECEIVING_EMAIL_OPTION:
+                $field_attrib = array_intersect_key($attrib, array_flip(array('id', 'class', 'style', 'cols', 'rows', 'tabindex', 'data-recipient-input'))) + array(
```

Comment:
See above: I don't understand what this extraction of HTML attributes out of `$attrib` means. When trying locally, `$attrib` contains these keys only: name, part, id, form, data-recipient-input, content & abort.